### PR TITLE
Harmonization of resource names for SAP HANA SR clusters

### PR DESCRIPTION
**Pull request by jankohoutek**
Harmonization of resource names for SAP HANA SR clusters

Must be merged togather with https://github.com/SUSE/qe-sap-deployment/pull/287

 Verification run:
  - https://openqaworker15.qa.suse.cz/tests/300016 - PASSED
  - https://openqaworker15.qa.suse.cz/tests/300017 - PASSED
  - https://openqaworker15.qa.suse.cz/tests/300018 - PASSED
  - https://openqaworker15.qa.suse.cz/tests/299517 - PASSED
  - https://openqaworker15.qa.suse.cz/tests/299518 - PASSED
  - https://openqaworker15.qa.suse.cz/tests/299519 - PASSED

### CODE REVIEW

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/sles4sap_publiccloud.pm [hunk 1]
```diff
@@ -129,11 +129,12 @@ sub run_cmd {
 
 sub get_promoted_hostname {
     my ($self) = @_;
+    my $master_resource_type = get_var('USE_SAP_HANA_SR_ANGI') ? "mst" : "msl";
```
Comment (alvarocarvajald) [inline lib/sles4sap_publiccloud.pm:132]: I don't see `mst_` resources in https://github.com/SUSE/qe-sap-deployment/pull/284. Are they coming later?

File: lib/sles4sap_publiccloud.pm [hunk 2]
```diff
@@ -298,11 +299,12 @@ sub is_hana_resource_running {
     my ($self, %args) = @_;
     $args{quiet} //= 0;
     my $hostname = $self->{my_instance}->{instance_id};
+    my $master_resource_type = get_var('USE_SAP_HANA_SR_ANGI') ? "mst" : "msl";
```
Comment (mpagot) [inline lib/sles4sap_publiccloud.pm:302]: This code looks like doing the same of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20401/files#diff-4b7291e584e1d0b487f685a7b45efc74dcd804ee2236acf7a866003d0f4fb87aR132

I suggest moving it in a `sub`

File: tests/publiccloud/sles4sap.pm [hunk 3]
```diff
@@ -208,10 +208,11 @@ sub run {
                 # Do the HANA "magic" if needed
                 if ($cluster_type eq 'hana') {
                     my $remoteHost = $hostname;
+                    my $master_resource_type = get_var('USE_SAP_HANA_SR_ANGI') ? "mst" : "msl";
```
Comment (mpagot) [inline tests/publiccloud/sles4sap.pm:211]: I can see this file only used in schedule `schedule/publiccloud/ha_sap.yml` 

It this schedule or test module still used anywhere? Is there a way to run a VR for this change? Is this test code related to qe-sap-deployment?

Comment (jankohoutek) [reply tests/publiccloud/sles4sap.pm:211]: DUNNO. It's another hard coded resource name and now it will work much cleaner

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: lib/sles4sap_publiccloud.pm (hunk 2 above)
Comment (mpagot) [reply lib/sles4sap_publiccloud.pm:302]: to keep this logic centralized, let say tomorrow we want to change setting name and/or triplet to use as name

**Pull request merged**